### DISCUSSION TITLE
Commits and Pull Request title convention

### DISCUSSION THREAD

**Discussion by eitsupi** [devcontainers-community/.github#4] [Ideas]
@jcbhmr For monorepo, my preference is to employ squash merge and conventional commit title to clarify which commit belongs to which component and reduce the number of commits. e.g. `feat(featurename): add a new option to ...`

What are your thoughts on repository settings?

![image](https://user-images.githubusercontent.com/50911393/226156463-9fd1e371-a36c-4c00-b1d6-b025fa4db65f.png)

**Comment by jcbhmr** [comment]
I'm open to any conventions! ♥️ If you want to take the initiative, go ahead. I'm more of a "refine your technique as you grow" (meaning I tend to only do conventions like that when I need to) kinda person, but I'll defer to you since you seem to know what you're talking about. 

In other words, go for any settings or convention you want 🚀 (just try to document them somewhere lmao 🧐)

**Reply by eitsupi** [reply]
Thanks for your opinion.

Perhaps it would be worth considering a voting system or something similar when the organization has more members.

**Reply by jcbhmr** [reply]
> Perhaps it would be worth considering a voting system or something similar when the organization has more members.

Good idea! 👍 Put it on the docket 😆

**Reply by eitsupi** [reply]
For now, I protected the main branch of https://github.com/devcontainers-community/templates and set only allow squash merging.

![image](https://user-images.githubusercontent.com/50911393/226184108-6b0c3268-228c-4d15-b6d3-17546ffffe91.png)
![image](https://user-images.githubusercontent.com/50911393/226184343-29bd4ea1-1190-4981-a6d6-dbf4b021f2bb.png)

If we wanted to restrict the PR title, we could use the following action.
https://github.com/amannn/action-semantic-pull-request
